hatch_pip_compile/plugin.py

```diff
@@ -347,3 +371,37 @@ def pipools_environment_dict(self) -> Dict[str, Any]:
         Get the environment dictionary
         """
         return self.metadata.hatch.config.get("envs", {})
+
+    def _piptools_install_dependencies(self) -> None:
+        """
+        Install dependencies with `pip-sync` or `pip`-install`
+        """
+        if self.install_method == "pip" and self.dependencies:
+            self._pip_install_cli()
+        elif self.install_method == "pip-sync":
+            self._pip_sync_cli()
+        else:
+            msg = f"Invalid pip-tools install method: {self.install_method}"
+            raise NotImplementedError(msg)
+
+    def install_project_no_deps(self) -> None:
+        """
+        Install the project (`--no-deps`)
+        """
+        with self.safe_activation():
+            proj_with_feats = self.apply_features(str(self.root))
+            self.platform.check_command(
+                self.construct_pip_install_command(args=["--no-deps", proj_with_feats])
```

Comment:
I didn't understand why it's OK to install the project with no deps.

Reply:
Aha OK because everything in `self.dependencies` has already been locked, and then the lock will be installed.

https://github.com/pypa/hatch/blob/fe663ba71f5223ce5850b65939183a5a8a9bf61f/src/hatch/env/plugin/interface.py#L338